react-client/src/Components/AnnotationPanel/AddAnnotationPanel.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect, useRef, useContext } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { AiOutlineClose } from "react-icons/ai";
+import GlobalContext from "../../store/global-context";
+import useLocalStorageStore from "../../hooks/useLocalStorageStore";
+import * as domUtils from "../../utils/domUtils";
+
+
+const AddAnnotationPanel = React.memo(({onRemoveAnnotation, onUpdatedAnnotation, targetElement}) => {  
+  const [txtInput, setTxtInput] = useState("");
+  const [displayInput, setDisplayInput] = useState(false);
+  const [annotationStore, setAnnotationStore] = useLocalStorageStore('annotation', []);
+  const txtInputRef = useRef(null);
+  
+  useEffect(() => {
+    setTimeout(() => {
+      txtInputRef.current.focus();
+    }, 300);
+    return () => {};
+  }, []);
+  
+  useEffect(() => {      
+    const hasExistingAnnotations = domUtils.hasAnnotations(annotationStore, targetElement);
+    if (hasExistingAnnotations === true) {
```

Comment:
Not a bug but you can also write it like this:

`if (hasExistingAnnotations) {`

Reply:
Because telling the code: `hasExistingAnnotations === true` is redundant. Your variable word itself `hasExistingAnnotations` already tells the whole story of your logic.